crates/core/tedge_agent/src/restart_manager/restart_operation_handler.rs

```diff
@@ -3,38 +3,24 @@ pub mod restart_operation {
     use crate::restart_manager::error::RestartManagerError;
     use camino::Utf8Path;
     use std::fs::File;
-    use std::fs::OpenOptions;
     use std::io::Read;
-    use std::io::Write;
     use time::OffsetDateTime;
 
     const TEDGE_AGENT_RESTART: &str = "tedge_agent_restart";
     const SLASH_PROC_UPTIME: &str = "/proc/uptime";
 
-    /// creates an empty file in /tmp
-    /// the file name defined by `TEDGE_AGENT_RESTART`
+    /// Creates a witness file that contains the time just before a reboot.
     ///
-    /// # Example
-    /// ```
-    /// RestartOperationHelper::create_tmp_restart_file()?;
-    /// ```
-    pub fn create_tmp_restart_file(tmp_dir: &Utf8Path) -> Result<(), RestartManagerError> {
+    /// This file will be compared on restart with /proc/uptime,
+    /// to determine if a reboot actually happened.
+    ///
+    /// See [has_rebooted]
+    pub async fn create_tmp_restart_file(tmp_dir: &Utf8Path) -> Result<(), RestartManagerError> {
         let path = &tmp_dir.join(TEDGE_AGENT_RESTART);
-        let path = Utf8Path::new(path);
+        let date_utc = OffsetDateTime::now_utc().unix_timestamp().to_string();
+
+        tedge_utils::fs::atomically_write_file_sync(path, date_utc.as_bytes())?;
```

Comment:
For a reason I fail to understand for now, using the async version of `atomically_write_file_sync` makes failing the test `restart_manager::tests::test_new_restart_operation`.

The function `atomically_write_file_sync` called to create and populate a `TEDGE_AGENT_RESTART` fails to open the file for write. This is not an access issue as this works with `std:fs:File`.

Reply:
Okay I got it. The `atomically_write_file_async` function is correct, but the test is based on a wrong assumption: i.e. that the restart witness file is create *before* is sent over MQTT the executing message.

=> I updated the restart actor so the executing message is sent only when all the files, used to check whether or not the device has reboot, are actually created.